// sh-ordinal-numbers/fesm2020/sh-ordinal-numbers.mjs
import * as i0 from '@angular/core';
import { Pipe, NgModule } from '@angular/core';

class OrdinalPipe {
    constructor() {
        this.ordinalVal = [
            'th',
            'st',
            'nd',
            'rd',
            'th',
            'th',
            'th',
            'th',
            'th',
            'th',
        ];
    }
    transform(value, args) {
        if (!value) {
            return '';
        }
        if (value > 9 && value < 20) {
            return value + this.ordinalVal[0];
        }
        const num = +String(value).slice(-1);
        return value + this.ordinalVal[num];
    }
}
OrdinalPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: OrdinalPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
OrdinalPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: OrdinalPipe, name: "ordinal" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: OrdinalPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'ordinal'
                }]
        }] });

class OrdinalNumbersModule {
}
OrdinalNumbersModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: OrdinalNumbersModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
OrdinalNumbersModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: OrdinalNumbersModule, declarations: [OrdinalPipe], exports: [OrdinalPipe] });
OrdinalNumbersModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: OrdinalNumbersModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: OrdinalNumbersModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        OrdinalPipe
                    ],
                    exports: [
                        OrdinalPipe
                    ]
                }]
        }] });

class ShOrdinalNumbersModule {
}
ShOrdinalNumbersModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: ShOrdinalNumbersModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ShOrdinalNumbersModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: ShOrdinalNumbersModule, exports: [OrdinalNumbersModule] });
ShOrdinalNumbersModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: ShOrdinalNumbersModule, imports: [[], OrdinalNumbersModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: ShOrdinalNumbersModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    exports: [
                        OrdinalNumbersModule
                    ]
                }]
        }] });

/*
 * Public API Surface of sh-ordinal-numbers
 */

/**
 * Generated bundle index. Do not edit.
 */

export { OrdinalNumbersModule, OrdinalPipe, ShOrdinalNumbersModule };
