// sh-ordinal-numbers/esm2020/lib/ordinal-numbers/pipe/ordinal.pipe.mjs
import { Pipe } from '@angular/core';
import * as i0 from "@angular/core";
export class OrdinalPipe {
    constructor() {
        this.ordinalVal = [
            'th',
            'st',
            'nd',
            'rd',
            'th',
            'th',
            'th',
            'th',
            'th',
            'th',
        ];
    }
    transform(value, args) {
        if (!value) {
            return '';
        }
        if (value > 9 && value < 20) {
            return value + this.ordinalVal[0];
        }
        const num = +String(value).slice(-1);
        return value + this.ordinalVal[num];
    }
}
OrdinalPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: OrdinalPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
OrdinalPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: OrdinalPipe, name: "ordinal" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: OrdinalPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'ordinal'
                }]
        }] });
